src/v/raft/consensus.cc

```diff
@@ -338,6 +338,7 @@ consensus::success_reply consensus::update_follower_index(
             idx.last_dirty_log_index = reply.last_dirty_log_index;
             idx.last_flushed_log_index = reply.last_flushed_log_index;
             idx.next_index = details::next_offset(idx.last_dirty_log_index);
+            idx.match_index = model::offset{};
```

Comment:
> Fixed problem that may lead to recovery stall

_What is the process_ that leads to the recovery stall?

The commit message describes what the code change is (offset moves backwards, follower match index is reset). That might be what leads to the stall, but it isn't clear why that causes a stall and why this fixes it.

Reply:
i was trying to reproduce this, this change will be included in follow up PR with changes in recovery stm,